src/fidesops/ops/service/strategy.py

```diff
@@ -0,0 +1,77 @@
+from __future__ import annotations
+
+import logging
+from abc import ABC
+from typing import Any, Dict, Generic, List, Optional, Type, TypeVar
+
+from pydantic import ValidationError
+
+from fidesops.ops.common_exceptions import NoSuchStrategyException
+from fidesops.ops.common_exceptions import ValidationError as FidesopsValidationError
+from fidesops.ops.schemas.saas.strategy_configuration import StrategyConfiguration
+
+logger = logging.getLogger(__name__)
+T = TypeVar("T", bound="Strategy")
+C = TypeVar("C", bound=StrategyConfiguration)
+
+
+def _find_strategy_subclass(
+    cls: Type[Strategy], strategy_name: str
+) -> Optional[Type[Strategy]]:
+    if hasattr(cls, "name") and cls.name == strategy_name:
+        return cls
+    for sub in cls.__subclasses__():
+        found = _find_strategy_subclass(sub, strategy_name)
+        if found is not None:
+            return found
+    return None
+
+
+def _find_all_strategy_subclasses(
+    cls: Type[T], subs: List[Type[T]] = None
+) -> List[Type[T]]:
+    if not subs:
+        subs = []
+    if hasattr(cls, "name"):
+        subs.append(cls)
+    for sub in cls.__subclasses__():
+        _find_all_strategy_subclasses(sub, subs)
```

Comment:
Your use of `extend` probably worked better from the previous commit, with this change, the return is prematurely ending the `for` loop.

Reply:
thank you for catching this!

it actually seems the issue was the `if not subs:` conditional - that evaluates to `True` in the case where `subs` is an empty list, and i was assuming it would evaluate to `False` (whoops). my next commit will update the conditional to `if subs is None:`, and this seems to be working as expected locally.

if you still think the `extend` implementation is better, than i'm happy to change it back. i think with the current approach here we avoid a bunch of overhead in creating temporary lists on each subcall in the recursion, and i also find this a bit more readable.

Reply:
Ah that makes sense, this implementation is fine!